### PR TITLE
Fix 3.9 build

### PR DESCRIPTION
**Pull request by erikseulean**
It seems like `Union[None, type]` is transformed into `Optional[type]`.

Signed-off-by: Erik-Cristian Seulean <eseulean@bloomberg.net>

### CODE REVIEW

**Comment by erikseulean** [comment]
@gaborbernat you can rebase on top of this and remove the 3.9 if.

**Comment by gaborbernat** [comment]
@erikseulean this would need to go on top of my fix commit, because this way the test suite is running on 3.8...

**Comment by erikseulean** [comment]
That sounds good, let's merge yours first.

**Comment by erikseulean** [comment]
Hmm, @gaborbernat I don't think the checks are running the correct version. At least the 2.7 one seems to be running on a 3.x I think.

File: src/attrs_strict/_type_validation.py [hunk 1]
```diff
@@ -179,7 +179,9 @@ def _handle_callable(attribute, callable_, expected_type):
         param.annotation for param in _signature.parameters.values()
     ]
     callable_args.append(_signature.return_annotation)
-    if not expected_type.__args__:  # type: ignore
+    if (
+        not hasattr(expected_type, "__args__") or not expected_type.__args__
+    ):  # type: ignore
```
Comment (gaborbernat) [inline src/attrs_strict/_type_validation.py:184]: cleaner to say:

```
if not getattr(expected_type, "__args__", None):
  return
```

File: src/attrs_strict/_type_validation.py [hunk 2]
```diff
@@ -179,7 +179,7 @@ def _handle_callable(attribute, callable_, expected_type):
         param.annotation for param in _signature.parameters.values()
     ]
     callable_args.append(_signature.return_annotation)
-    if not expected_type.__args__:  # type: ignore
+    if getattr(expected_type, "__args__", None):  # type: ignore
```
Comment (gaborbernat) [inline src/attrs_strict/_type_validation.py:182]: ``if not``

**Pull request merged**